src/connector/src/sink/iceberg/jni_catalog.rs

```diff
@@ -0,0 +1,262 @@
+                )
+                .set_source(e)
+                .with_context("table_name", &table_name_str)
+            })?;
+
+        let rust_json_str = unsafe {
+            let jvalue = result_json.as_jni();
+            let jstring = JString::from_raw(jvalue.l);
+            let java_str = env.get_string(&jstring).map_err(|e| {
+                icelake::Error::new(
+                    ErrorKind::Unexpected,
+                    "Failed to get string from return value of loadTable method of jni catalog",
+                )
+                .set_source(e)
+            })?;
+
+            java_str
+                .to_str()
+                .map_err(|e| {
+                    icelake::Error::new(
+                        ErrorKind::Unexpected,
+                        "Failed to convert java string to rust utf8 string",
+                    )
+                    .set_source(e)
+                    .with_context("Java string bytes", format!("{:?}", java_str.to_bytes()))
+                })?
+                .to_string()
+        };
+
+        let resp: LoadTableResult = serde_json::from_str(&rust_json_str)?;
+
+        let metadata_location = resp.metadata_location.clone().ok_or_else(|| {
+            icelake::Error::new(
+                icelake::ErrorKind::IcebergFeatureUnsupported,
+                "Loading uncommitted table is not supported!",
+            )
+        })?;
+
+        tracing::info!("Table metadata location of {table_name} is {metadata_location}");
+
+        let table_metadata = resp.table_metadata()?;
+
+        let iceberg_io_args = IcebergTableIoArgs::builder_from_path(&table_metadata.location)?
+            .with_args(self.config.table_io_configs.iter())
+            .build()?;
+        let table_op = iceberg_io_args.create()?;
+
+        Ok(
+            Table::builder_from_catalog(table_op, self.clone(), table_metadata, table_name.clone())
+                .build()?,
+        )
+    }
+
+    async fn update_table(self: Arc<Self>, update_table: &UpdateTable) -> icelake::Result<Table> {
+        let _guard = self.jvm.attach_current_thread().map_err(|e| {
+            icelake::Error::new(
+                ErrorKind::Unexpected,
+                "Failed to attach current thread to jvm",
+            )
+            .set_source(e)
+        })?;
+
+        let request_str = serde_json::to_string(&CommitTableRequest::try_from(update_table)?)?;
+
+        let mut env = self.jvm.get_env().map_err(|e| {
+            icelake::Error::new(ErrorKind::Unexpected, "Failed to get jni env").set_source(e)
+        })?;
+
+        let request_jni_str = env.new_string(&request_str).map_err(|e| {
+            icelake::Error::new(
+                ErrorKind::Unexpected,
+                "Failed to create jni string from request json",
+            )
+            .set_source(e)
+            .with_context("request", &request_str)
+        })?;
+
+        let result_json = env
+            .call_method(
```

Comment:
We can use the `call_method` macro to avoid manually generating the the method signature and casting the return type.